File: 2707-merge-two-2d-arrays-by-summing-values/2707-merge-two-2d-arrays-by-summing-values.java
class Solution {
    public int[][] mergeArrays(int[][] nums1, int[][] nums2) {
       HashMap<Integer,Integer> map=new HashMap<>();

       for(int i=0; i<nums2.length; i++) map.put(nums2[i][0],nums2[i][1]);

       for(int i=0;  i<nums1.length; i++){
        if(map.containsKey(nums1[i][0])){
            int x=map.get(nums1[i][0]);
            map.put(nums1[i][0],nums1[i][1]+x);
        }else
            map.put(nums1[i][0],nums1[i][1]);
       }
       int[][] ans=new int[map.size()][2];
       int i=0;
       for(Map.Entry<Integer,Integer> x: map.entrySet()){
            ans[i][0]=x.getKey();
            ans[i][1]=x.getValue();
            i++;
       }
       Arrays.sort(ans,(a,b)->Integer.compare(a[0],b[0]));
       return ans;
    }
}

File: 2707-merge-two-2d-arrays-by-summing-values/2707-merge-two-2d-arrays-by-summing-values.py
class Solution:
    def mergeArrays(self, n1: List[List[int]], n2: List[List[int]]) -> List[List[int]]:
        ans=[]
        nl1,nl2,i,j=len(n1),len(n2),0,0

        while i<nl1 and j<nl2:
            if n1[i][0]==n2[j][0]:
                ans.append([n1[i][0],n1[i][1]+n2[j][1]])
                i +=1
                j +=1
            elif n1[i][0]<n2[j][0]:
                ans.append([n1[i][0],n1[i][1]])
                i +=1
            else:
                ans.append([n2[j][0],n2[j][1]])
                j +=1
        
        while i<nl1:
            ans.append(n1[i])
            i+=1
        while j<nl2:
            ans.append(n2[j])
            j+=1
        
        return ans
        